sdk/core/core/inc/az_span.h

```diff
@@ -20,93 +20,94 @@
 typedef struct {
   struct {
     uint8_t * begin;
-    size_t capacity;
-    size_t length;
+    int32_t length;
+    int32_t capacity;
   } _internal;
 } az_span;
 
-AZ_NODISCARD AZ_INLINE az_span az_span_empty() { return (az_span){ 0 }; }
+/********************************  SPAN GETTERS / SETTERS */
 
-AZ_NODISCARD AZ_INLINE bool az_span_is_empty(az_span const span) {
-  return span._internal.length == 0;
+/**
+ * @brief convenient way to get the capacity from one span
+ *
+ */
+AZ_NODISCARD AZ_INLINE int32_t az_span_capacity_get(az_span const span) {
+  return span._internal.capacity;
 }
 
 /**
- * @brief returns a span with the left @var n bytes of the given @var span.
+ * @brief convenient way to set the capacity from one span
  *
- * If the @var n is greater than the span capacity then the whole @var span is returned.
  */
-AZ_NODISCARD AZ_INLINE az_span az_span_take(az_span const span, size_t const n) {
-  if (span._internal.capacity <= n) {
-    return span;
-  }
-  return (az_span){ ._internal = { .begin = span._internal.begin, .capacity = n, .length = n } };
+AZ_NODISCARD AZ_INLINE void az_span_capacity_set(az_span span, int32_t const capacity) {
+  span._internal.capacity = capacity;
 }
 
 /**
- * @brief returns a span with @var n positions are dropped.
+ * @brief convenient way to get the length from one span
  *
- * If the @var n is greater than span capacity then an empty span is returned
  */
-AZ_NODISCARD AZ_INLINE az_span az_span_drop(az_span const span, size_t const n) {
-  if (span._internal.capacity <= n) {
-    return az_span_empty();
-  }
-  size_t const length_and_capacity = span._internal.length - n;
-  return (az_span){ ._internal = { .begin = span._internal.begin + n,
-                                   .capacity = length_and_capacity,
-                                   .length = length_and_capacity } };
+AZ_NODISCARD AZ_INLINE int32_t az_span_length_get(az_span const span) {
+  return span._internal.length;
 }
 
 /**
- * Returns a sub span of the given span.
+ * @brief convenient way to set the length from one span
+ *
  */
-AZ_NODISCARD AZ_INLINE az_span
-az_span_sub(az_span const span, size_t const begin, size_t const end) {
-  az_span const t = az_span_take(span, end);
-  return az_span_drop(t, begin);
+AZ_NODISCARD AZ_INLINE void az_span_length_get(az_span span, int32_t const length) {
```

Comment:
Accept by constant pointer to non-constant, as I suggested above; otherwise setting length isn't going to do anything.
Also, change the name to az_span_length_set.
And also now I am not sure if we should allow users to directly modify (at least, increase) length and capacity.
If should be safe to decrease length, relatively safe to decrease capacity, but I am not sure if we want to allow to increase capacity that easily. Shouldn't you initialize a new span to do that? It could point to the same region, but writing that code would make user to check correctness more, probably. Also, when you increase length - there should be append() that adds something to the buffer and increases the length, but as far as increasing the length without appending anything, I am not sure.